### PR TITLE
[REEF-780]  Lower down the log level when pulling HttpEndPoint to red…

### PR DESCRIPTION
**Pull request by jwang98052**
…uce noise and call SubmitAndGetDriverUrl to match the code at Java side.

JIRA: [REEF-780](https://issues.apache.org/jira/browse/REEF-780)

This closes #

Author: Julia Wang  Email: juliaw@apache.org

### CODE REVIEW

**Comment by afchung** [comment]
I will take a pass.

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java [hunk 1]
```diff
@@ -225,7 +225,7 @@ private void writeDriverHttpEndPoint(final File driverFolder,
     String trackingUri = null;
     for (int i = 0; i < 60; i++) {
       try {
-        LOG.log(Level.INFO, "Attempt " + i + " reading " + httpEndpointPath.toString());
+        LOG.log(Level.FINE, "Attempt " + i + " reading " + httpEndpointPath.toString());
         if (fs.exists(httpEndpointPath)) {
           FSDataInputStream input = fs.open(httpEndpointPath);
```
Comment (afchung) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: final

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java [hunk 2]
```diff
@@ -225,7 +225,7 @@ private void writeDriverHttpEndPoint(final File driverFolder,
...
           BufferedReader reader = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8));
```
Comment (afchung) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: final

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java [hunk 3]
```diff
@@ -225,7 +225,7 @@ private void writeDriverHttpEndPoint(final File driverFolder,
     String trackingUri = null;
```
Comment (afchung) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: It's probably worth it to put start and end messages that are on the INFO level.

**Comment by afchung** [comment]
@jwang98052 Just a few minor comments. Thanks!

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java (hunk 1 above)
Comment (jwang98052) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: Done

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java (hunk 2 above)
Comment (jwang98052) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: Done

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java (hunk 3 above)
Comment (jwang98052) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: Done

**Comment by jwang98052** [comment]
Review comments addressed and changes pushed in.

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java [hunk 4]
```diff
@@ -223,14 +223,16 @@ private void writeDriverHttpEndPoint(final File driverFolder,
     final Path httpEndpointPath = new Path(dfsPath, fileNames.getDriverHttpEndpoint());
 
     String trackingUri = null;
+    LOG.log(Level.INFO, "Attempt to reading " + httpEndpointPath.toString());
```
Comment (afchung) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: @jwang98052 Can you add an end comment for the successful scenario?

**Comment by afchung** [comment]
@jwang98052 Just one more comment. Thanks!

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java (hunk 4 above)
Comment (jwang98052) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: I already have LOG.log(Level.INFO, "Successfully reading trackingUri :" + trackingUri); after successfully read and before break from the loop. 
If after 60 loops, it is still not able to read, this line will be logged 
 LOG.log(Level.WARNING, "Failed reading " + httpEndpointPath.toString());

**Comment by afchung** [comment]
@jwang98052 I see, I missed it earlier. In that case, note that `readLine()` could return `null` if end of stream has been reached, as documented here:
http://docs.oracle.com/javase/7/docs/api/java/io/BufferedReader.html
Could you modify it to print the message after the null comparison? After that, I will test and merge. Thanks!

File: lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java [hunk 5]
```diff
@@ -223,14 +223,16 @@ private void writeDriverHttpEndPoint(final File driverFolder,
...
-        LOG.log(Level.INFO, "Attempt " + i + " reading " + httpEndpointPath.toString());
+        LOG.log(Level.FINE, "Attempt " + i + " reading " + httpEndpointPath.toString());
         if (fs.exists(httpEndpointPath)) {
-          FSDataInputStream input = fs.open(httpEndpointPath);
-          BufferedReader reader = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8));
+          final FSDataInputStream input = fs.open(httpEndpointPath);
+          final BufferedReader reader = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8));
           trackingUri = reader.readLine();
           reader.close();
+          LOG.log(Level.INFO, "Successfully reading trackingUri :" + trackingUri);
```
Comment (afchung) [inline lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java]: Please move to after comparison with `null`.

**Comment by jwang98052** [comment]
Change pushed in.

**Pull request closed without merging**